packages/fxa-content-server/app/tests/spec/lib/survey-targeter.js

```diff
@@ -0,0 +1,97 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import SurveyTargeter from 'lib/survey-targeter';
+import { assert } from 'chai';
+import sinon from 'sinon';
+
+import storageMock from '../../mocks/storage-mock';
+
+const sandbox = sinon.createSandbox();
+const trueFn = sandbox.stub().returns(true);
+const nullFn = sandbox.stub().returns(null);
+
+describe('lib/SurveyTargeter', () => {
+  let surveyTargeter;
+  const doNotBotherSpan = 2592000000;
+  const config = {
+    enabled: true,
+    doNotBotherSpan,
+  };
+  const surveys = [
+    {
+      id: 'portugese-speaking-mobile-users-in-southern-hemisphere',
+      conditions: { relier: null },
+      view: 'settings',
+      rate: 0.1,
+      url: 'https://www.surveygizmo.com/s3/5541940/pizza',
+    },
+  ];
+
+  beforeEach(() => {
+    sandbox.resetHistory();
+    sandbox.restore();
+    surveyTargeter = new SurveyTargeter({
+      window: {
+        localStorage: storageMock(),
+        navigator: {
+          userAgent:
+            'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:76.0) Gecko/20100101 Firefox/76.0',
+        },
+      },
+      relier: {
+        get: nullFn,
+      },
+      user: {
+        getSignedInAccount: trueFn,
+      },
+      config,
+      surveys,
+    });
+  });
+
+  it('surveyTargeter returns a survey if the view matches', async () => {
+    const view = await surveyTargeter.getSurvey('settings');
+    assert.equal(view.el.className, 'survey-wrapped');
+  });
+
+  it('surveyTargeter sets "lastSurvey" in localStorage if view returns', async () => {
+    const setItemSpy = sandbox.spy(surveyTargeter._storage, 'set');
+    await surveyTargeter.getSurvey('settings');
+    const lastSurveyValue = surveyTargeter._storage.get('lastSurvey');
+
+    assert.isNumber(lastSurveyValue);
+    assert.isTrue(setItemSpy.called);
+
+    const args = setItemSpy.getCalls()[0].args;
+    assert.equal(args[1], lastSurveyValue);
```

Comment:
The assertions can be shortened to `sinon.assert.calledOnceWithExactly(setItemSpy, 'lastSurvey', lastSurveyValue);`